sources/packages/backend/apps/queue-consumers/src/processors/schedulers/institution-integration/ier12-integration/_tests_/e2e/ier12-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,394 @@
+    );
+  });
+
+  it("Should generate an IER12 file with one record for a student with one pending disbursement due to a restriction.", async () => {
+    // Arrange
+    const testInputData = {
+      student: JOHN_DOE_FROM_CANADA,
+      application: {
+        applicationNumber: defaultApplicationNumber,
+        studentNumber: "1",
+        relationshipStatus: RelationshipStatus.Single,
+        applicationStatus: ApplicationStatus.Completed,
+        applicationStatusUpdatedOn: undefined,
+      },
+      assessment: {
+        triggerType: AssessmentTriggerType.OriginalAssessment,
+        assessmentDate: undefined,
+        workflowData: WORKFLOW_DATA_SINGLE_INDEPENDENT_WITH_NO_DEPENDENTS,
+        assessmentData: ASSESSMENT_DATA_SINGLE_DEPENDENT,
+        disbursementSchedules: [
+          {
+            coeStatus: COEStatus.completed,
+            disbursementScheduleStatus: DisbursementScheduleStatus.Pending,
+            disbursementDate: undefined,
+            updatedAt: undefined,
+            dateSent: undefined,
+            disbursementValues: AWARDS_SINGLE_DISBURSEMENT,
+          },
+        ],
+      },
+      educationProgram: PROGRAM_GRADUATE_DIPLOMA_WITH_INSTITUTION_PROGRAM_CODE,
+      offering: OFFERING_FULL_TIME,
+    };
+    const application = await saveIER12TestInputData(db, testInputData, {
+      programYearPrefix: sharedProgramYearPrefix,
+      submittedDate: referenceSubmissionDate,
+    });
+    // Finds a restriction that stops the funding.
+    const stopFullTimeDisbursementRestriction = await db.restriction.findOne({
+      where: {
+        actionType: ArrayContains([
+          RestrictionActionType.StopFullTimeDisbursement,
+        ]),
+      },
+    });
+    // Saves the restrictions that will force the record status to be set to DISR.
+    await saveFakeStudentRestriction(db.dataSource, {
+      student: application.student,
+      application,
+      restriction: stopFullTimeDisbursementRestriction,
+    });
+
+    // Queued job.
+    const job = createIER12SchedulerJobMock(
+      application.currentAssessment.assessmentDate,
+    );
+
+    // Act
+    const ier12Results = await processor.processIER12File(job);
+    db.application.createQueryBuilder();
+
+    // Assert
+    const uploadedFile = getUploadedFile(sftpClientMock);
+    // Assert process result.
+    expect(ier12Results).toBeDefined();
+    // File timestamp.
+    const [timestampResult] = getFileNameAsCurrentTimestampMock.mock.results;
+    expect(isValidFileTimestamp(timestampResult.value)).toBe(true);
+    expect(ier12Results).toStrictEqual([
+      {
```

Comment:
getSuccessSummaryMessages() is not used here ?

Reply:
Just missed this one. I will change it.

Reply:
Changed.